charts/spring-service/templates/deployment.yaml

```diff
@@ -71,34 +71,41 @@ spec:
             - name: 'SENTRY_DSN'
               valueFrom:
                 secretKeyRef:
-                  name: sentry
-                  key: "dsn-{{ .Values.serviceName }}"
+                  name: {{ .Values.serviceName }}-external-secret
+                  key: "sentry/dsn-{{ .Values.serviceName }}"
             {{- if .Values.env.springDatasourceFromSecret }}
             - name: "SPRING_DATASOURCE_URL"
               valueFrom:
                 secretKeyRef:
-                  name: {{ .Values.env.springDatasourceFromSecret }}
-                  key: jdbc-url
+                  name: {{ .Values.serviceName }}-external-secret
+                  key: "{{ .Values.env.springDatasourceFromSecret }}/jdbc-url"
```

Comment:
Slash im Key? Ich fände ein `-` besser, das ist eher mit CLI tools verträglich.

Man kann auch einfach als Key den Env-Name nehmen, dann ist es frei von Sonderzeichen und klar worauf des später gemappt wird.

Edit: Dem Code nach zu urteilen, sollte da auch bereits der Env-Name sein, oder?

Reply:
ich hatte ursprünglich den Env-Namen, fand es dann aber eher verwirrend, weil dann nicht ganz klar, was jeweils gemeint ist -> die Env oder das Secret, das in die Env reingereicht würde
dass CLI-Tools mit Slashes in Identifiern Probleme hätten verstehe ich nicht wirklich. Sonst dürften wir eigentlich weder Secrets Stores noch Dateisysteme in unseren Containern benutzen?

Reply:
ich hab ja gestern mit yq migriert, da waren später auch slashes drin (weil ich werte umbenennen wollte), und es gab kein Problem. Ich muss aber zugeben, ich hätte auch nicht gedacht, dass es funktioniert.

Also bindestrich hilft nicht oder? Weil in .Values.env.springDatasourceFromSecret sind auch noch slashes drin?
Ich finde TKs Vorschlag hier so gut, habe aber auch nicht die super harte Meinung...

Reply:
Ich hatte am Anfang auch noch name und key vertauscht.
Den Key ist ja eigentlich echt auch nur für k8s, um Secret auf Deployment zu mappen. Falls es doch noch Probleme macht, können wirs ja ändern, da es ja dann hermetisch pro Helm-Deploy ist.